Doc/c-api/type.rst

```diff
@@ -117,6 +117,13 @@ Type Objects
    If no module is associated with the given type, sets :py:class:`TypeError`
    and returns ``NULL``.
 
+   This function is usually used to get the module in wich a method is defined.
+   Note that in such a method, ``PyType_GetModule(Py_TYPE(self))``
+   may not return the intended result.
+   ``Py_TYPE(self)`` may be a *subclass* of the intended class, and subclasses
+   are not necessarily defined in the same module as their superclass.
+   See :c:type:`PyCMethod` to get the defining class.
```

Comment:
The "defining class" is something new to me. PyCMethod mentions it without explining it. Would it be possible to define it somewhere? And then link to it.

Reply:
Do you think the explanation should be in `PyCMethod` docs, or in `METH_METHOD | METH_FASTCALL | METH_KEYWORDS` docs?
For the other conventions, the docs for the flags have more detail. But I haven't found a way to link to `METH_METHOD | METH_FASTCALL | METH_KEYWORDS`.